### PR TITLE
Macroize the common `ReportBody` member `Accessor`s

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/1" target="_blank">Graphite</a> 
    * **PR #2** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/2" target="_blank">Graphite</a> 
      * **PR #3** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/3" target="_blank">Graphite</a> 
        * **PR #9** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/9" target="_blank">Graphite</a> 
          * **PR #10** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/10" target="_blank">Graphite</a> 
            * **PR #11** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/11" target="_blank">Graphite</a> 
              * **PR #12** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/12" target="_blank">Graphite</a> 
              * **PR #18** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/18" target="_blank">Graphite</a> 
                * **PR #20** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/20" target="_blank">Graphite</a>  👈
    * **PR #4** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/4" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/20?utm_source=stack-comment).

File: verifier/src/report_body.rs [hunk 1]
```diff
@@ -24,16 +24,35 @@ pub trait Accessor<T>: Debug {
...
+                self.$accessor_method()
+            }
+        }
+
+        impl Accessor<$member_type> for $member_type {
+            fn get(&self) -> $member_type {
+                self.clone()
+            }
+        }
+    )*}
```
Comment (nick-mobilecoin) [inline verifier/src/report_body.rs:49]: I ended up using a generic in #26

File: verifier/src/report_body.rs [hunk 2]
```diff
@@ -24,16 +24,35 @@ pub trait Accessor<T>: Debug {
...
-        self.attributes()
-    }
+/// Macro to generate boilerplate for implementing [`Accessor`] for a member of
+/// [`ReportBody`].
+///
+/// Will create [`Accessor`] implementations for both `ReportBody` and the
+/// underlying member type
+///
+/// # Arguments
+/// * `member_type` - The type of the member
```
Comment (samdealy) [inline verifier/src/report_body.rs:34]: nit: Can you clarify what `member` refers to here?

**Review by samdealy — APPROVED** [review]

**Review by jcape — APPROVED** [review]

File: verifier/src/report_body.rs (hunk 2 above)
Comment (nick-mobilecoin) [reply verifier/src/report_body.rs]: I took a stab at this, but I'm up for suggestions on more clarity.
I renamed to  "field" as that seemed to be used more often in rust lingo.
It feels a little awkward as none of the fields of `ReportBody` are public so they're only accessible through methods on `ReportBody`

**Review by samdealy — APPROVED** [review]

**Pull request merged**